include/logger/sampleRecord.h

```diff
@@ -121,6 +121,7 @@ void free_sample_buffer(struct sample *s);
  * @return true if the sample was found and set
  */
 bool get_sample_value_by_name(const struct sample *s, const char * name, double *value, char ** units);
+bool get_channel_value_by_name(const char * name, double *value, char ** units);
```

Comment:
I would refactor everywhere throughout the firmware and switch to just using get_channel_value_by_name. The original function is no longer needed. 

This is great overall!!

Reply:
I did try this on my first pass through, but felt like I was tugging on a thread that might unravel if I went any further.  See: logger_sample_cb_t and camera_control_sample_cb.  These callbacks require the sample parameter.  I didn't actually check to see if these were used with other functions, but I assumed that if they were needed in the first place they would take more than one value.